src/vmm/src/builder.rs

```diff
@@ -371,6 +382,32 @@ pub(crate) fn setup_interrupt_controller(
         .map_err(StartMicrovmError::Internal)
 }
 
+/// Sets up the serial device.
+pub fn setup_serial_device(
+    event_manager: &mut EventManager,
+    input: Box<dyn devices::legacy::ReadableFd + Send>,
+    out: Box<dyn io::Write + Send>,
+) -> std::result::Result<Arc<Mutex<Serial>>, StartMicrovmError> {
+    let interrupt_evt = EventFd::new(libc::EFD_NONBLOCK)
+        .map_err(Error::EventFd)
+        .map_err(StartMicrovmError::Internal)?;
+    let protected_serial = EventManager::wrap_handler(devices::legacy::Serial::new_in_out(
```

Comment:
Nit: Unnecessary overhead in naming the serial as `protected_serial`, `serial` would do just fine.